packages/jest-jasmine2/src/__tests__/each.test.js

```diff
@@ -0,0 +1,266 @@
+            'expected string',
+            expectFunction,
+          );
+          expect(globalMock).toHaveBeenCalledWith(
+            'expected string',
+            expectFunction,
+          );
+        });
+
+        test('calls global function with title containing param values when using sprintf format', () => {
+          const environmentMock = getEnvironmentMock();
+          installEach(environmentMock);
+
+          const globalMock = environmentMock.global[keyPath];
+
+          globalMock.each([['hello', 1], ['world', 2]])(
+            'expected string: %s %s',
+            noop,
+          );
+
+          expect(globalMock).toHaveBeenCalledTimes(2);
+          expect(globalMock).toHaveBeenCalledWith(
+            'expected string: hello 1',
+            expectFunction,
+          );
+          expect(globalMock).toHaveBeenCalledWith(
+            'expected string: world 2',
+            expectFunction,
+          );
+        });
+
+        test('calls global function with cb function containing all parameters of each test case', () => {
+          const environmentMock = getEnvironmentMock();
+          installEach(environmentMock);
+
+          const globalMock = environmentMock.global[keyPath];
+          const testCallBack = jest.fn();
+          globalMock.each([['hello', 'world'], ['joe', 'bloggs']])(
+            'expected string: %s %s',
+            testCallBack,
+          );
+
+          globalMock.mock.calls[0][1]();
+          expect(testCallBack).toHaveBeenCalledTimes(1);
+          expect(testCallBack).toHaveBeenCalledWith('hello', 'world');
+
+          globalMock.mock.calls[1][1]();
+          expect(testCallBack).toHaveBeenCalledTimes(2);
+          expect(testCallBack).toHaveBeenCalledWith('joe', 'bloggs');
+        });
+
+        test('calls global function with async done when cb function has more args than params of given test row', () => {
+          expect.hasAssertions();
+          const environmentMock = getEnvironmentMock();
+          installEach(environmentMock);
+
+          const globalMock = environmentMock.global[keyPath];
+          globalMock.each([['hello']])('a title', (hello, done) => {
+            expect(hello).toBe('hello');
+            expect(done).toBe('DONE');
+          });
+
+          globalMock.mock.calls[0][1]('DONE');
+        });
+      });
+
+      describe('Table Tagged Template Literal', () => {
+        test('throws error when there are fewer arguments than headings when given one row', () => {
+          const environmentMock = getEnvironmentMock();
+          installEach(environmentMock);
+
+          const globalMock = environmentMock.global[keyPath];
+          const testCallBack = jest.fn();
+
+          globalMock.each`
+            a    | b    | expected
+            ${0} | ${1} |
+          `('this will blow up :(', testCallBack);
+
+          expect(() =>
+            globalMock.mock.calls[0][1](),
+          ).toThrowErrorMatchingSnapshot();
```

Comment:
instead of doing this, I wonder if it would be better to create actual integration tests (in https://github.com/facebook/jest/tree/master/integration-tests) and assert on the output there?

While this works, it seems sorta weird to invoke something passed to a mock function, and making asserts on how that behaves.

Might just be me, though!

Reply:
I agree it would be nice to have an integration test too, I used TDD when building this thats why it seems strange but it allowed me to just write unit tests for the behaviour 😃 

I'll add some tests in /integration-tests too (I didn't know about that directory)